low-level/framework/utils/consulstore.py

```diff
@@ -32,18 +33,29 @@ class ConsulStore(Store):
 
     def __init__(self, host, port):
         super(Store, self).__init__()
+        self._data_sync_required = False
+        self._consul_conn_status = False
+        self._host = host
+        self._port = port
+        self._establish_consul_connection(self._host, self._port)
+        self._file_store = FileStore()
+
+    def _establish_consul_connection(self, host, port):
         for retry_index in range(0, MAX_CONSUL_RETRY):
             try:
                 self.consul_conn = consul.Consul(host=host, port=port)
+                self._consul_conn_status = True
                 break
 
             except requests.exceptions.ConnectionError as connerr:
                 logger.warn("Error[{0}] consul connection refused Retry Index {1}" \
                     .format(connerr, retry_index))
+                self._consul_conn_status = False
                 time.sleep(WAIT_BEFORE_RETRY)
 
             except Exception as gerr:
                 logger.warn("Error[{0}] consul error".format(gerr))
```

Comment:
No, We do retry in case of the consul connection error only. Actually, It was implemented to handle failover case when consul  would not be available for the short span of time and We use to get consul connection refused error.

Reply:
Yes. In the current implementation, We are not doing retries in case other generic exception. We do retires only for the connection error only. What do you think ? Should we treat all the consul related exception as  connection error and do the retries ?